Merge some of the WordPress lint rules, to reduce number of reported issues

wzieba opened the pull request:
### Description

This PR disables `UnkownNullness` lint check, to not duplicate lint warnings of the same type we get from WordPress-Lint.

### Testing

Not necessary. 

Related: https://github.com/wordpress-mobile/WordPress-Lint-Android/pull/20

wzieba committed:
temp: use wip version of wordpress lint checks

dangermattic commented:
<!--
  0 Errors
  1 Warning: PR is not assigned to a milest...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

ParaskP7 commented on config/lint/lint.xml:
```diff
@@ -79,4 +79,6 @@
     <!-- TODO: https://github.com/wordpress-mobile/WordPress-Android/issues/21065 -->
     <issue id="UsingMaterialAndMaterial3Libraries" severity="warning" />
+    <!--  We rely on more granural checks from WordPress-Lint-Android  -->
+    <issue id="UnknownNullness" severity="ignore" />
```
Suggestion (💡): Instead of ignoring `UnknownNullness` I actually recommend we revert #19116 and remove all the configuration related to that, effectively removing it, not ignoring. 🤔

wzieba replied:
Thank you @ParaskP7 ! I've applied your suggestion and rebranded this PR to not introduce changed WordPress Lint checks. I don't have capacity to work on this now, but I think PR in this shape will also be useful to merge!

wzieba committed:
Revert "temp: use wip version of wordpress lint checks"

This reverts commit e210382d828308718f704a14744fabd6cb8ef5b5.

wzieba committed:
Merge branch 'trunk' into bump_android_lint

wzieba committed:
Don't enable `UnknownNullness` lint warning

We have this covered via checks from WordPress-Lint

wzieba committed:
Merge branch 'trunk' into bump_android_lint

ParaskP7 commented on libs/login/build.gradle:
```diff
@@ -18,7 +18,6 @@ android {
     }
     lint {
-        enable += 'UnknownNullness'
         disable 'UseTomlInstead'
```
Suggestion (💡): I understand that we could remove this here too, now that all dependencies within this module are migrated to version catalogs. And then get rid of the whole `lint { ... }` block, just like you did for the rest of the modules.

PS: We could potentially do the same for the `fluxc` module, but we would first need to migrate the below two dependencies to version catalogs (however, I am not sure about the `IgnoreWithoutReason`):

```groovy
    api "com.goterl:lazysodium-android:5.0.2@aar"
    api "net.java.dev.jna:jna:5.15.0@aar"
```

wzieba replied:
> PS: We could potentially do the same for the fluxc module, but we would first need to migrate the below two dependencies to version catalogs

This is not easy to do though, because of the aar specification not working as intended. This is needed, because both of these dependencies offer both `aar` and `jar` and one of them even defaults to `jar`. This seems to be a known issue at Gradle (https://github.com/gradle/gradle/issues/21267) but not yet fixed.

But regarding the `login` module, I agree, I'll fix this in a sec!

ParaskP7 replied:
> But regarding the login module, I agree, I'll fix this in a sec!

Thanks @wzieba ! 🙏 

> This seems to be a known issue at Gradle (https://github.com/gradle/gradle/issues/21267) but not yet fixed.

Yea, true, I tried really quick myself and couldn't get it working (the right way), but I didn't thought this would be such a blocker, interesting, thanks for sharing this issue... 🤔 

I guess, in the meanwhile, we could do the following, adding a TODO comment point to the issue you shared:

`libs.version.toml`
```toml
...
[versions]
terl-lazysodium-android = '5.0.2@aar'
jna = '5.15.0@aar'
...
[libraries]
terl-lazysodium-android = { module = "com.goterl:lazysodium-android", version.ref = "terl-lazysodium-android" }
jna = { module = "net.java.dev.jna:jna", version.ref = "jna" }
...
```

`build.gradle`
```groovy
api(libs.terl.lazysodium.android.get().toString()) // TODO: https://github.com/gradle/gradle/issues/21267
api(libs.jna.get().toString()) // TODO: https://github.com/gradle/gradle/issues/21267
```

Wdyt? 🤔

wzieba replied:
This looks neat, but I'm unable to verify that `aar` instead of `jar` is actually used. 

Let's take the JNA dependency for the example: it offers both `aar` and `jar`. The default artifact specified in the pom file is `jar`: https://mvnrepository.com/artifact/net.java.dev.jna/jna/5.15.0

When I applied your suggestion and when I run `:WordPress:dependencyInsight --dependency net.java.dev.jna:jna --configuration wordpressWasabiDebugRuntimeClasspath`, it seems that `jar` is used. You can compare this result to result of any other dependency, e.g. `:WordPress:dependencyInsight --dependency androidx.credentials:credentials --configuration wordpressWasabiDebugRuntimeClasspath` which prints usage of `aar`.

But what's strange, when I run the dependency insight for JNA on `trunk`, I also see that `jar` is requested 😕 .

For this reason, as I'm not entirely sure how it all work, I'd postpone changing this dependencies to version catalog as I'm not sure what would be the effect. Happy to hear your thoughts - maybe there's a way to make sure `aar` is used?

ParaskP7 replied:
@wzieba , I did a simple `println(...)` to verify the exact string that is used. In the end, using this `.get().toString()` trick should give us exactly what we had before, a simple string, there shouldn't be any more magic involved, not having version catalogs choosing between `aar` and `jar` whatsoever (🤔). This is my assumption at least. 🤞 

```groovy
println("LazySodium: ${libs.terl.lazysodium.android.get().toString()}")
println("JNA: ${libs.jna.get().toString()}")
```

wzieba replied:
Gotcha 👍 when I removed `@aar` the project won't compile, and with the code you suggested, build is complete so I think we're all good 👍 

I've also applied changed to remove `IgnoreWithoutReason` from disabled rules.

ParaskP7 reviewed (approved):
👋 @wzieba !

I have reviewed this PR as per the instructions, everything works as expected, good job! 🌟 

-----

I have left one suggestion (💡) for you to consider. I am going to approve this PR anyway, since none is blocking. I am **NOT** going to merge this PR yet to give you some time to apply any of my suggestions. However, feel free to ignore them and merge the PR yourself.

wzieba committed:
Disable disabling `UseTomlInstead` in `login` module

wzieba committed:
Merge branch 'trunk' into bump_android_lint

wzieba committed:
Use version catalog for dependencies that specify artifact type

Related discussion: https://github.com/wordpress-mobile/WordPress-Android/pull/21256#discussion_r1827550283

wzieba committed:
Remove `UseTomlInstead` from disabled lint rules

wzieba committed:
Remove `IgnoreWithoutReason` from disabled lint rules

Suppress this warning on imported tests that are breaking this rule

wzieba committed:
Merge branch 'trunk' into bump_android_lint
